src/MSBuild/XMake.cs

```diff
@@ -1296,7 +1296,8 @@ string[] commandLine
 
                         // InvalidProjectFileExceptions and its aggregates have already been logged.
                         if (exception is not InvalidProjectFileException
-                            && !(exception is AggregateException aggregateException && aggregateException.InnerExceptions.All(innerException => innerException is InvalidProjectFileException)))
+                            && !(exception is AggregateException aggregateException && aggregateException.InnerExceptions.All(innerException => innerException is InvalidProjectFileException))
```

Comment:
```suggestion
                            && !(exception is AggregateException aggregateException && aggregateException.InnerExceptions.All(innerException => innerException is InvalidProjectFileException || innerException is CircularDependencyException))
```
?

Reply:
It actually has to be `&& exception is not CircularDependencyException`

Reply:
You already added that one; I meant within the AggregateException. I imagine it's ok if it's an aggregation of InvalidProjectFileExceptions and CircularDependencyExceptions, right?

Reply:
I don't think a `InvalidProjectFileException` and `CircularDependencyException` are both possible at the same time. I think all projects are evaluated and then as a separate pass the edges are added and that's where circular dependencies are detected. The 1st pass would error with `InvalidProjectFileException`s